src/coi.rs

```diff
@@ -1,28 +1,35 @@
 use crate::coi_message_filter::CoiMessageFilter;
 use crate::context::*;
-use crate::error::Error;
 use crate::imap::*;
+use crate::job::*;
+use crate::param::*;
+
+const COI_METADATA_ENABLED: &str = "/private/vendor/vendor.dovecot/coi/config/enabled";
+const COI_METADATA_MESSAGE_FILTER: &str =
+    "/private/vendor/vendor.dovecot/coi/config/message-filter";
 
 impl Context {
     pub fn get_coi_config(&self) -> Option<CoiConfig> {
         self.inbox.read().unwrap().get_coi_config()
     }
 
-    pub fn set_coi_enabled(&self, enable: bool) -> Result<(), Error> {
-        self.inbox.write().unwrap().set_coi_enabled(self, enable)
+    pub fn set_coi_enabled(&self, enable: bool, id: i32) {
+        let value = if enable { "yes" } else { "" };
```

Comment:
We have to pass Some("yes"), and None instead of "" AFAIK, or is "" treated the same like NIL?

Reply:
Ah, it's using `Params`. I wonder how you distinguish a "" from a NIL.

Reply:
At the moment, we don't. I think omitting `=` in the serialization would be a way to add support for that. But since empty values are not a valid for any of metadata we use so far, I think we can deal with that later.